sdks/v4-sdk/src/utils/encodeRouteToPath.ts

```diff
@@ -0,0 +1,32 @@
+import { Currency } from '@uniswap/sdk-core'
+import { Route } from '../entities/route'
+import { ADDRESS_ZERO } from './internalConstants'
+
+export type PathKey = {
+  intermediateCurrency: string // address
+  fee: number
+  tickSpacing: number
+  hooks: string // address
+  hookData: string // bytes
+}
+
+export const encodeRouteToPath = (route: Route<Currency, Currency>): PathKey[] => {
+  let currencyIn = route.input
+  let pathKeys: PathKey[] = []
+
+  for (let pool of route.pools) {
+    const currencyOut = currencyIn.equals(pool.currency0) ? pool.currency1 : pool.currency0
+
+    pathKeys.push({
+      intermediateCurrency: currencyOut.isNative ? ADDRESS_ZERO : currencyOut.address,
```

Comment:
we do need to pass in `exactOutput`, just like v3 `encodeRouteToPath` input function params. The reason is because in case of exact out, intermediateCurrency will be currencyIn for current for-loop iteration.